### PR TITLE
feat: operator to extract text in images using tesseract

### PR DESCRIPTION
**Pull request by aatmanvaidya**
Adding a new operator to detect and extract text in images using tesseract

### CODE REVIEW

**Comment by aatmanvaidya** [comment]
@dennyabrain while reviewing the PR can you also check if the file nomenclature is correct or not?

File: src/api/core/operators/test_detect_text_in_image_tesseract.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        # initialize operator
+        detect_text_in_image_tesseract.initialize(param={})
+    
+    @classmethod
+    def tearDownClass(cls):
+        # delete config files
+        pass
+
+    def test_sample_image_from_disk(self):
+        image_path = "sample_data/text.png"
```
Comment (dennyabrain) [inline src/api/core/operators/test_detect_text_in_image_tesseract.py:16]: @aatmanvaidya looks like the text depends on a file stored on disk? Can we add this to the repo too? you can create the `sample_data` folder in the `operators` folder itself.

File: src/api/core/operators/test_detect_text_in_image_tesseract.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        image_text = detect_text_in_image_tesseract.run(image_path)
+        expected_text = "It was the best of\ntimes, it was the worst\nof times, it was the age\nof wisdom, it was the\nage of foolishness..."
```
Comment (dennyabrain) [inline src/api/core/operators/test_detect_text_in_image_tesseract.py:18]: @aatmanvaidya lets use a sample image with multilingual or non english text.

File: src/api/core/operators/test_detect_text_in_image_tesseract.py (hunk 1 above)
Comment (aatmanvaidya) [reply src/api/core/operators/test_detect_text_in_image_tesseract.py:16]: so in the other test files, the `sample_data` folder is just refereed, and it was also there in the .gitignore file, that's why I didn't add it, but I can add it now

File: src/api/core/operators/test_detect_text_in_image_tesseract.py (hunk 2 above)
Comment (aatmanvaidya) [reply src/api/core/operators/test_detect_text_in_image_tesseract.py:18]: will make the change

File: src/api/core/operators/test_detect_text_in_image_tesseract.py (hunk 1 above)
Comment (dennyabrain) [reply src/api/core/operators/test_detect_text_in_image_tesseract.py:16]: got it. yeah I think with just me working on it, that was fine but lets include the sample_data folder to the repo to reduce the steps to setup feluda locally. As long as we are careful about not uploading a large 10 gb video file in it, we should be fine.

**Comment by aatmanvaidya** [comment]
@dennyabrain I have made the changes, let me know if any more changes are needed

**Pull request merged**